engine/runtime/src/main/scala/org/enso/compiler/Compiler.scala

```diff
@@ -252,13 +265,22 @@ class Compiler(
 
   private def recognizeBindings(
     module: IR.Module,
-    moduleContext: ModuleContext
+    moduleContext: ModuleContext,
+    isGeneratingDocs: Boolean
```

Comment:
this is wrong – instead of running a different set of passes, just add your pass into `moduleDiscoveryPasses`. Give it a config to control whether or not it does anything. Pass the config variable inside `moduleContext`.